File: frontend/src/types/job.ts
/**
 * Job Type Definitions
 */

export interface Job {
  id: string;
  title: string;
  description: string;
  location: string;
  salary_min?: number;
  salary_max?: number;
  required_skills: string[];
  recruiter_id: string;
  is_active: boolean;
  created_at: string;
  updated_at?: string;
}

export interface JobCreate {
  title: string;
  description: string;
  location: string;
  salary_min?: number;
  salary_max?: number;
  required_skills: string[];
}

export interface JobSearchParams {
  skip?: number;
  limit?: number;
  location?: string;
  title?: string;
}

File: backend/app/schemas/job.py
"""
Job Schemas
"""

from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from decimal import Decimal


class JobBase(BaseModel):
    title: str
    description: str
    location: str
    salary_min: Optional[Decimal] = None
    salary_max: Optional[Decimal] = None
    required_skills: List[str] = []


class JobCreate(JobBase):
    pass


class JobUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    location: Optional[str] = None
    salary_min: Optional[Decimal] = None
    salary_max: Optional[Decimal] = None
    required_skills: Optional[List[str]] = None


class JobResponse(JobBase):
    id: str
    recruiter_id: str
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

